message.go

```diff
@@ -322,6 +324,12 @@ func (m *Message) asyncRead(onComplete func(error)) io.WriteCloser {
 	reader, writer := io.Pipe()
 	m.reader = reader
 	go func() {
+		defer func() {
+			if p := recover(); p != nil {
+				log.Printf("PANIC in BLIP asyncRead: %v\n%s", p, debug.Stack())
```

Comment:
Add reader close here (with nil check)

Reply:
I'm uncertain of the desired behavior, or side effects of closing the reader. I'm hesitant to make this change until I understand this more.